apps/admin/backend/src/store.ts

```diff
@@ -105,6 +105,22 @@ export class Store {
     return new Store(DbClient.fileClient(dbPath, SchemaPath));
   }
 
+  /**
+   * Runs the given function in a transaction. If the function throws an error,
+   * the transaction is rolled back. Otherwise, the transaction is committed.
+   *
+   * You can also control rollback by including a `shouldCommit` callback.
+   *
+   * Returns the result of the function.
+   */
+  withTransaction<T>(
+    fn: () => Promise<T>,
+    shouldCommit?: (result: T) => boolean
+  ): Promise<T>;
+  withTransaction<T>(fn: () => T, shouldCommit?: (result: T) => boolean): T {
+    return this.client.transaction(() => fn(), shouldCommit);
+  }
```

Comment:
I think it's safe to assume that throwing an exception or returning an `err` should roll back. Accepting a `shouldCommit` parameter feels a bit too flexible for our use cases